app/repositories/course_reserve_repository.rb

```diff
@@ -0,0 +1,131 @@
+          courses_json
+        )
+      )
+    end
+
+    private
+
+      def courses_json
+        courses_data.encode('UTF-8', 'binary', invalid: :replace, undef: :replace, replace: '?')
+      end
+
+      def courses_data
+        Faraday.get("#{ENV['bibdata_base']}/courses").body
+      end
+  end
+
+  class Relation
+    attr_reader :courses
+    attr_accessor :query_params
+    delegate :map, to: :to_a
+    def initialize(courses, query_params = {})
+      self.courses = courses
+      self.query_params = query_params
+    end
+
+    def solr_query
+      QueryBuilder.new(query_params).to_s
+    end
+
+    class QueryBuilder
+      attr_reader :instructor, :course, :department
+      def initialize(query_params)
+        self.instructor = query_params.fetch(:instructor, nil)
+        self.course = query_params.fetch(:course_with_id, nil)
+        self.department = query_params.fetch(:department_with_identifier, nil)
+      end
+
+      def instructor=(instructor)
+        return if instructor.blank?
+        @instructor = instructor
+      end
+
+      def course=(course)
+        return if course.blank?
+        @course = course
+      end
+
+      def department=(department)
+        return if department.blank?
+        @department = department
+      end
+
+      def to_s
+        (['type_s:ReserveListing'] + [:instructor, :course, :department].map do |x|
+          to_query(x)
+        end).compact.join(' AND ')
+      end
+
+      private
+
+        def to_query(field)
+          value = send(field)
+          return unless value
+          "#{field}_s:\"#{value}\""
+        end
+    end
+
+    def query(query_params)
+      self.class.new(
+        select do |course|
+          query_params.map do |key, value|
+            value.blank? || course.send(key) == value
+          end.uniq == [true]
+        end.to_a,
+        query_params.merge(self.query_params)
+      )
+    end
+
+    def to_a
+      @to_a ||= courses
+    end
+
+    def select(&block)
+      self.class.new(to_a.select(&block))
+    end
+
+    def instructors
+      @instructors ||= courses.map(&:instructor).uniq
+    end
+
+    def departments
+      @departments ||= courses.map(&:department_with_identifier).uniq
+    end
+
+    def course_names
+      @course_names ||= courses.map(&:course_with_id).uniq
+    end
+
+    def reserve_list_ids
+      @reserve_list_ids ||= courses.map(&:reserve_list_id).uniq
+    end
+
+    private
+
+      def courses=(courses)
+        @courses = courses.map { |x| Course.new(*x.values) }
+      end
+  end
+
+  Course = Struct.new(:reserve_list_id, :department_name, :department_code, :course_name,
+                      :course_number, :section_id, :instructor_first_name,
+                      :instructor_last_name) do
+    attr_accessor :bib_ids
+    def instructor
+      "#{instructor_last_name}, #{instructor_first_name}"
+    end
+
+    def department_with_identifier
+      "#{department_code}:#{department_name}"
```

Comment:
Can you add a space after the colon? I don't know why Voyager doesn't add the space, but adding would be consistent with the course list dropdown.